tools_docker/Debian_New_Uwsgi/Dockerfile

```diff
@@ -1,134 +1,117 @@
+RUN apt-get install -qy $(cat "${PACKAGE_DIR}/run_backend_packages.list" "${PACKAGE_DIR}/optional_preview_packages.list") \
+    && apt-get autoremove -qy \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/ /tmp/* /var/tmp/*
 
-RUN if ([ "$TAG" != "" ] && [ "$BRANCH" != "" ]); \
-    then echo "Its not possible to use ARG TAG and ARG BRANCH in same time." && exit 1;fi
 
-# Install required packages
-RUN \
-    apt install -qy curl \
-    # Node.JS 14
-    && curl -sL https://deb.nodesource.com/setup_14.x | bash - \
-    && apt install -qy \
-        build-essential \
-        libjpeg-dev \
-        libmagickwand-dev \
-        libpq-dev \
-        nodejs \
-        python3-dev \
-        python3-pip \
-        python3-venv
+###
+# Backend install stage
+####
+# Stage which:
+# - installs python dependencies
+# - installs tracim backend
+# - installs optional dependencies
+####
+FROM base_install AS build_backend
+ARG PACKAGE_DIR
+# required packages for building the backend dependencies
+RUN apt-get install -qy $(cat "${PACKAGE_DIR}/build_backend_packages.list") \
+    # Go in backend folder
+    && cd /tracim/backend \
+    # Install backend
+    && pip install --force-reinstall \
+    -r requirements-build.txt \
+    -r requirements.txt \
+    -r requirements-db-postgres.txt \
+    -r requirements-db-mysql.txt \
+    -r requirements-storage-s3.txt \
+    -r requirements-full-preview-generator.txt \
+    -e "."
 
-# Clone repo
-RUN \
-    get_source_from_github() { \
-    if ([ "$TAG" != "" ] && [ "$BRANCH" = "" ]); \
-    then git clone -b $TAG --single-branch "$REPO" tracim; \
-    elif ([ "$TAG" = "" ] && [ "$BRANCH" != "" ]); \
-    then git clone "$REPO" tracim && cd tracim/ && git checkout $BRANCH && cd -; \
-    else git clone "$REPO" tracim; \
-    fi \
-    } \
-    && get_source_from_github $TAG $BRANCH
 
+###
+# Frontend build stage
+####
+# Stage which:
+# - installs javascript dependencies
+# - setups backend translation
+# - builds frontend
+####
+FROM node:14-buster AS build_frontend
+COPY --from=base_install /tracim /tracim
 RUN \
-# Go in backend folder
     cd /tracim/backend \
-# Install backend
-    && python3 -m pip install -r requirements-build.txt \
-    && python3 -m pip install -r requirements.txt \
-    && python3 -m pip install -r requirements-full-preview-generator.txt \
-    && python3 -m pip install -r requirements-db-postgres.txt \
-    && python3 -m pip install -r requirements-db-mysql.txt \
-    && python3 -m pip install -e "." \
     && npm install "i18next-conv@<8" -g \
     && ./update_i18n_json_file.sh
-
+# Install frontend
 RUN \
-# Go in tracim folder
     cd /tracim/ \
-# Install frontend
     && ./install_frontend_dependencies.sh root \
     && ./build_full_frontend.sh
-
-RUN \
 # Clean frontend installation
+RUN \
     rm -rf /tracim/frontend*/node_modules \
     && rm -rf /tracim/node_modules
 
 
 FROM base_install
 
-MAINTAINER contact@tracim.fr
+LABEL org.opencontainers.image.authors="contact@tracim.fr"
```

Comment:
is there other metadata we should provide in docker, if so, maybe adding a specific issue ?